pkg/http/api.go

```diff
@@ -1920,17 +1917,17 @@ func (a *api) onPostStateTransaction(reqCtx *fasthttp.RequestCtx) {
 	storeName := reqCtx.UserValue(storeNameParam).(string)
 	store, ok := a.universal.CompStore.GetStateStore(storeName)
 	if !ok {
-		err := messages.ErrStateStoreNotFound.WithFormat(storeName)
+		err := apierrors.StateStoreNotFound(storeName)
 		log.Debug(err)
 		universalFastHTTPErrorResponder(reqCtx, err)
 		return
 	}
 
 	transactionalStore, ok := store.(state.TransactionalStore)
-	if !ok {
-		msg := NewErrorResponse("ERR_STATE_STORE_NOT_SUPPORTED", fmt.Sprintf(messages.ErrStateStoreNotSupported, storeName))
-		fasthttpRespond(reqCtx, fasthttpResponseWithError(nethttp.StatusInternalServerError, msg))
-		log.Debug(msg)
+	if !ok || !state.FeatureTransactional.IsPresent(store.Features()) {
```

Comment:
A conformance test in components-contrib is in place to ensure that if `TransactionalStore` is implemented, the feature is exposed, and vice-versa.

```suggestion
	if !ok {
```

Reply:
We needed this here because when we're using a pluggable component, the interface check will always be satisfied.